src/core/thread/network_data_local.cpp

```diff
@@ -181,13 +181,13 @@ otError Local::AddService(uint32_t       aEnterpriseNumber,
     VerifyOrExit(mLength + sizeof(NetworkDataTlv) + serviceTlvLength <= sizeof(mTlvs), error = OT_ERROR_NO_BUFS);
 
     serviceTlv = reinterpret_cast<ServiceTlv *>(mTlvs + mLength);
-    Insert(reinterpret_cast<uint8_t *>(serviceTlv), serviceTlvLength + sizeof(NetworkDataTlv));
+    Insert(reinterpret_cast<uint8_t *>(serviceTlv), static_cast<uint8_t>(serviceTlvLength) + sizeof(NetworkDataTlv));
```

Comment:
verify `serviceTlvLength` is less than 255 or static assert `sizeof(mTlvs) < 255`?